pyfar/dsp/filter/frequency_weighting.py

```diff
@@ -0,0 +1,370 @@
+"""
+Functions and helpers to create frequency weighting filters
+according to IEC 61672-1.
+"""
+import warnings
+from typing import Callable, Literal, Optional
+import numpy as np
+import pyfar as pf
+import pyfar.constants as pfc
+import scipy.signal as sps
+import scipy.optimize as spo
+from pyfar.constants.frequency_weighting import _F_1, _F_2, _F_3, _F_4
+
+
+_A_WEIGHTING_ZEROS = [0, 0, 0, 0]
+_A_WEIGHTING_POLES = [_F_1, _F_1, _F_4, _F_4, _F_2, _F_3]
+_C_WEIGHTING_ZEROS = [0, 0]
+_C_WEIGHTING_POLES = [_F_1, _F_1, _F_4, _F_4]
+
+
+def frequency_weighting_filter(
+        signal,
+        target_weighting: Literal["A", "C"]="A",
+        n_frequencies=100,
```

Comment:
in `Signal` & `FrequencyData` we use `n_bins` instead of `n_frequencies`, which could be considered here.